RESOURCE_ACCESS_CONTROL_FOR_PLUGINS.md

```diff
@@ -0,0 +1,493 @@
+        return new SampleResourceParser();
+    }
+}
+```
+
+
+### **Calling Access Control Methods from the ResourceSharingClient Client**
+Plugins must **declare a dependency** on `opensearch-security-client` and use it to call access control methods.
+The client provides **four access control methods** for plugins. For detailed usage and implementation, refer to the [`opensearch-security-client` README.md](./client/README.md)
+
+
+Tip: Refer to the `org.opensearch.sample.resource.client.ResourceSharingClientAccessor` class to understand the client setup in further detail.
+
+Example usage:
+```java
+ @Override
+void doExecute(Task task, ShareResourceRequest request, ActionListener<ShareResourceResponse> listener) {
+    if (request.getResourceId() == null || request.getResourceId().isEmpty()) {
+        listener.onFailure(new IllegalArgumentException("Resource ID cannot be null or empty"));
+        return;
+    }
+
+    ResourceSharingClient resourceSharingClient = ResourceSharingClientAccessor.getResourceSharingClient(nodeClient, settings);
+    resourceSharingClient.shareResource(
+            request.getResourceId(),
+            RESOURCE_INDEX_NAME,
+            request.getShareWith(),
+            ActionListener.wrap(sharing -> {
+                ShareResourceResponse response = new ShareResourceResponse(sharing.getShareWith());
+                listener.onResponse(response);
+            }, listener::onFailure)
+    );
+}
+```
+
+
+---
+
+## **5. What are Scopes?**
+
+This feature introduces a new **sharing mechanism** called **scopes**. Scopes define **the level of access** granted to users for a resource. They are **defined and maintained by plugins**, and the security plugin does **not** interpret or enforce their specific meanings. This approach gives plugins the **flexibility** to define scope names and behaviors based on their use case.
+
+Each plugin must **document its scope definitions** so that users understand the **sharing semantics** and how different scopes affect access control.
+
+Scopes enable **granular access control**, allowing shareableResources to be shared with **customized permission levels**, making the system more flexible and adaptable to different use cases.
+
+### **Common Scopes for Plugins to declare**
+| Scope        | Description                                         |
+|-------------|-----------------------------------------------------|
+| `PUBLIC`    | The resource is accessible to all users.            |
+| `READ_ONLY` | Users can view but not modify the resource.         |
+| `READ_WRITE` | Users can view and modify the resource.             |
+
+By default, all shareableResources are private and only visible to the owner and super-admins. Resources become accessible to others only when explicitly shared.
+
+SPI provides you an interface, with two default scopes `PUBLIC` and `READ_ONLY`, which can be extended to introduce more plugin-specific values.
```

Comment:
The table lists a scope `READ_WRITE` which is not mentioned in any other place. Is this on purpose?

Additionally, could you give an example for a more evolved set of scopes a real-world plugin might use?

I am asking because I am a bit lost what scopes can be. To a certain extent because `PUBLIC` and `READ_ONLY` seem to refer to different dimensions in my perception. `PUBLIC` refers to group of users, while `READ_ONLY` refers to an allowed action.

Reply:
these names currently serve semantic purpose only. The context and meaning of these scopes is left to the plugins to determine how they would like to define and utilize them. However, before we proceed on this discussion, let finalize the part forward for scopes as there seems to be confusion around it: https://github.com/opensearch-project/security/pull/5185#issuecomment-2744341082

Reply:
scopes have been renamed to action-groups. I've updated this section.